src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -1690,6 +1690,15 @@ export type Task = StudyTask | CompanyWorkTask | CreateProgramWorkTask | CrimeTa
  * @public
  */
 export interface Singularity {
+  /**
+   * This function returns the save data. If your browser supports Compression Streams API, the save data is a
```

Comment:
Can we get rid of this caveat? Converting a string to a Uint8Array is not very hard. Does this have to do with the underlying format being compressed vs. not?

Reply:
> Does this have to do with the underlying format being compressed vs. not?

Yes. The data is different:
- `Uint8Array`: compressed json save string. This is our current format.
- `string`: base64-encoded json save string. This is the legacy format. We still export the save data to this format if the player's browser does not support Compression Streams API.

Reply:
IIRC, we support (and will continue to support) the old format in perpetuity. The issue here is the different data types, which are a needless complication for the user (also I would assume for us, but maybe we are using it to differentiate between format types?)

IMO for an API we should be guaranteeing a single data type. Since the string is already base64-encoded (and thus ASCII), this should be as simple as `new TextEncoder().encode(str)`

Reply:
> maybe we are using it to differentiate between format types?

That's right. I'm not sure if we can "simplify" `export type SaveData = string | Uint8Array;` to `export type SaveData = Uint8Array;` and change related code. I'll take a look at it later. If there is no problem, I'll make a PR for it.

> IMO for an API we should be guaranteeing a single data type. Since the string is already base64-encoded (and thus ASCII), this should be as simple as new TextEncoder().encode(str)

I agree.